File: src/algorithmic_toolbox/week3/MaxAdvertRevenue.java
package algorithmic_toolbox.week3;

import java.util.Arrays;
import java.util.Scanner;

public class MaxAdvertRevenue {

  public static long getMaxRevenue(int[] profits, int[] clicks) {
    Arrays.sort(profits);
    Arrays.sort(clicks);

    long total = 0;
    for (int i = 0; i < profits.length; i++) {
      total += profits[i] * clicks[i]; 
    }
    return total;
  }
  public static void main(String[] args) {
    Scanner scanner = new Scanner(System.in);
    int N = scanner.nextInt();

    int[] profits = new int[N];
    int[] clicks = new int[N];

    for (int i = 0; i < N; i++) {
      profits[i] = scanner.nextInt();
    }
    for (int i = 0; i < N; i++) {
      clicks[i] = scanner.nextInt();
    }
    scanner.close();

    System.out.println(getMaxRevenue(profits, clicks));
  }
}

File: src/algorithmic_toolbox/week3/max_advert_revenue.py
from typing import List


def get_max_revenue(profits: List[int], clicks: List[int]) -> int:
    profits.sort()
    clicks.sort()
    
    total = 0
    for i in range(len(profits)):
        total += profits[i] * clicks[i]
    
    return total


if __name__ == '__main__':
    n = input()  # not required for Python
    profits = [int(x) for x in input().split()]
    clicks = [int(x) for x in input().split()]

    print(get_max_revenue(profits, clicks))
